### PR TITLE
feat: add WalletConnect v2 support to Ledger connector

### PR DESCRIPTION
**Pull request by hlopes-ledger**
## Description

Adds support for WalletConnect v2 to the Ledger connector.

The Ledger Connect Kit is currently in beta, will update package.json to the final one when released. No changes expected to the connector until the final CK release.

The update enables new options for WalletConnect v2

- walletConnectVersion?: 1 | 2
- projectId?: EthereumProviderOptions['projectId']
- requiredChains?: number[]
- requiredMethods?: string[]
- optionalMethods?: string[]
- requiredEvents?: string[]
- optionalEvents?: string[]

Support for WalletConnect v1 is still kept as it can be used by specifying a custom bridge URL.

To test it

- update the connector instance on the demo with the new options

        walletConnectVersion: 2,
        requiredChains: [1],
        projectId: 'a test id',

- Ledger Extension on Safari
    - Open the example application on Safari iOS or macOS
    - follow instructions to install and enable the extension
    - connect using the Extension
- WalletConnect and Ledger Live
    - Open the example application on Chrome iOS or macOS
    - install Ledger Live if not installed yet using the install link
    - connect with Ledger Live using the QR code or the button

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

### CODE REVIEW

File: packages/connectors/src/ledger.ts [hunk 1]
```diff
@@ -14,17 +15,32 @@ import {
...
+  enableDebugLogs?: boolean
+  walletConnectVersion?: 1 | 2
+
+  // WalletConnect v2 init parameters
+  projectId?: EthereumProviderOptions['projectId']
+  requiredChains?: number[]
+  requiredMethods?: string[]
+  optionalMethods?: string[]
+  requiredEvents?: string[]
+  optionalEvents?: string[]
```
Comment (jxom) [inline packages/connectors/src/ledger.ts:31]: Can we use a discrimitive union for these? ie:

```
type LedgerConnectorOptions = {
  walletConnectVersion: 1,
  ... // v1 opts
} | {
  walletConnectVersion: 2,
  ... // v2 opts
}
```

Comment (jxom) [reply packages/connectors/src/ledger.ts:31]: Actually, should we even support v1 in this PR at all if it's being turned off next week?

Comment (hlopes-ledger) [reply packages/connectors/src/ledger.ts:31]: From what I've seen on the community you'll still be able to use v1 with a custom bridge URL, by specifying it in the bridge parameter. We intend to keep it for now and move to v2 only at a later stage.

**Comment by hlopes-ledger** [comment]
We've released Connect Kit 1.1.0 and I've updated the package dependency.

**Review by jxom — APPROVED** [review]

**Pull request merged**